Update routing at homepage for seekers and owners

nadavsu opened the pull request:
# Description
The homepage needs to be rendered differently for each user type, so I created the initial routing function at `main/views.py`
`views.py` now routes the user based on his account type.
The different rendering functions still need to be developed.

## Manual Tests
Logging in with a user account with an apartment (owner) - displays `home.html`
Logging in with a user account without an apartment (shouldn't be done) - displays `landing_page.html`
Logging in with a seeker account - displays `home.html`
Logged out user - `landing_page.html`
This is good for now as we don't have the rendering functions for each user type.

### Issues closed by this PR
No issues closed by this PR yet as this is an intial PR for features that are under development.
This PR was needed for @tamirho and I to begin working on the owner and seeker homepages respectively.

nadavsu committed:
Update routing at homepage for seekers and owners

`views.py` now routes the user based on his account type.
This is just the skeleton, we still need to add the functions which we will route to.

Micha546 commented on main/views.py:
```diff
@@ -1,10 +1,20 @@
+    		function that will render the homepage for the owner.'''
+    	return render(request, 'main/home.html')
+    else:
+    	# Owners who have not added an apartment yet (ie not completed registration) can get here.
```
Do you mean someone who did not register to the site yet?

looks great 🔥

nadavsu replied:
Users who have registered as owners but didn't add an apartment yet can reach this condition.
This can happen when users request the homepage during the registration phase I believe.
Either way, nobody should get there. It should return the landing page or an error page.

Micha546 replied:
Our user creation form and seeker/apartment creation forms show up together, you can't create an account that is neither a Seeker or an Apartment

Micha546 replied:
The form creates a user only if the user and the apartment forms are valid

Micha546 replied:
So I think that the comment is wierd here, if theres a way to create a user that is not an apartment then it's a bug

nadavsu replied:
I see.
It's always good to keep that `else` statement, you never know how users can get to that condition, and if it's not handled then we'll have an error.

Micha546 replied:
I agree, it's just that I think that the comment above is a bit weird

nadavsu replied:
You're right. I'll remove it.

tamirho replied:
Maybe we should have an option for admin/staff user?

tamirho commented:
LGTM

simzacks commented on main/views.py:
```diff
@@ -2,9 +2,18 @@
 
 def home(request):
-    if request.user.is_authenticated:
+    if not request.user.is_authenticated:
```
You should use the @login_required decorator instead of an if not authenticated

tamirho replied:
I think that the `@login_required` redirects the user to the `home` view.
So it will create a recursive loop with no ending point.

nadavsu replied:
While I agree that we should use `@login_required` throughout the project, I don't think that's the case here.
I want to display the both the landing page and the homepage under the `''` (base) URL.
If I use the decorator, then I wouldn't be able to do that since I would need to redirect to a page which has a different URL.
I mean I can use use the decorator, but it comes with a cost of weird URL patterns like `127.0.0.1/home` for the homepage or `127.0.0.1/welcome` for the welcome page.

nadavsu replied:
Exactly what Tamir said.

nadavsu replied:
I added the decorator for the `home` function.

nadavsu committed:
Add login_required decorator for homepage

Added an index page function which routes the user to the homepage in case he is already logged in
The `home` function routes the user to the relevant homepage rendering function. The `home` function has a `login_required` decorator because only logged in users should be able to access the homepage.
Logged out users who will try to access the homepage will get redircted to the landing page.

tamirho reviewed (approved):
Good job
